runtimes/core/src/api/schema/cookie.rs

```diff
@@ -0,0 +1,343 @@
+
+            if let Some(c) = jar.get(cookie_name) {
+                // Only consider non-empty values to be present.
+                if !c.value().is_empty() {
+                    return true;
+                }
+            }
+        }
+        false
+    }
+
+    pub fn fields(&self) -> impl Iterator<Item = (&String, &jsonschema::Field)> {
+        self.schema.root().fields.iter()
+    }
+
+    pub fn parse_incoming_request_parts(
+        &self,
+        req: &axum::http::request::Parts,
+    ) -> APIResult<Option<PValues>> {
+        self.parse_req(&req.headers)
+    }
+
+    pub fn parse_req(&self, headers: &axum::http::HeaderMap) -> APIResult<Option<PValues>> {
+        if self.schema.root().is_empty() {
+            return Ok(None);
+        }
+
+        let mut jar = cookie::CookieJar::new();
+        for raw in headers.get_all(COOKIE) {
+            if let Ok(raw) = raw.to_str() {
+                for c in cookie::Cookie::split_parse(raw).flatten() {
+                    jar.add_original(c.into_owned());
+                }
+            }
+        }
+
+        match self.schema.parse(jar) {
+            Ok(decoded) => Ok(Some(decoded)),
+            Err(err) => Err(err),
+        }
+    }
+
+    pub fn parse_resp(&self, headers: &axum::http::HeaderMap) -> APIResult<Option<PValues>> {
+        if self.schema.root().is_empty() {
+            return Ok(None);
+        }
+
+        let mut jar = cookie::CookieJar::new();
+        for raw in headers.get_all(SET_COOKIE) {
+            let c = cookie::Cookie::parse(String::from_utf8_lossy(raw.as_bytes()))
+                .map_err(api::Error::internal)?;
+            jar.add_original(c.into_owned());
+        }
+
+        match self.schema.parse(jar) {
+            Ok(decoded) => Ok(Some(decoded)),
+            Err(err) => Err(err),
+        }
+    }
+}
+
+impl ToResponse for Cookie {
+    fn to_response(
+        &self,
+        payload: &super::JSONPayload,
+        mut resp: axum::http::response::Builder,
+    ) -> APIResult<axum::http::response::Builder> {
+        if self.schema.root().is_empty() {
+            return Ok(resp);
+        }
+
+        let Some(payload) = payload else {
+            return Err(api::Error {
+                code: api::ErrCode::InvalidArgument,
+                message: "missing cookie parameters".to_string(),
+                internal_message: Some("missing cookie parameters".to_string()),
+                stack: None,
+                details: None,
+            });
+        };
+
+        let schema = self.schema.root();
+        for (key, value) in payload.iter() {
+            let key = key.as_str();
+            let Some(field) = schema.fields.get(key) else {
+                continue;
+            };
+
+            let name = field.name_override.as_deref().unwrap_or(key);
+            let c = to_cookie(name, value)?;
+            resp = resp.header(SET_COOKIE, c.to_string());
+        }
+
+        Ok(resp)
+    }
+}
+
+struct RawCookie<'a> {
```

Comment:
How is this used? I was hoping we could keep the JavaScript-specific encoding in the JS runtime crate, if possible

Reply:
Ah, forgot about that when I was refactoring, will have a look if I can remove that, It was used internally when parsing a cookie out of a PValue

Reply:
I've removed this now. The problem is that when we parse pvalues from the js handler we dont have the schema, so there is no way of knowing if it should be a PValue::Object or PValue::Cookie. So I added that we pass the response schema down, and then we transform the parsed PValues based on that schema. Let me know if you think this is a good approach or if you have something else in mind